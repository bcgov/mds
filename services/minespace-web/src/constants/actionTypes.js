// Redux Actions
export const REQUEST = "REQUEST";
export const SUCCESS = "SUCCESS";
export const ERROR = "ERROR";
export const CLEAR = "CLEAR";

// Stores
export const STORE_USER_MINE_INFO = "STORE_USER_MINE_INFO";
export const STORE_MINE = "STORE_MINE";
export const STORE_MINE_DOCUMENTS = "STORE_MINE_DOCUMENTS";
export const STORE_MINE_REPORT_DEFINITION_OPTIONS = "STORE_MINE_REPORT_DEFINITION_OPTIONS";

// Authentication
export const AUTHENTICATE_USER = "AUTHENTICATE_USER";
export const LOGOUT = "LOGOUT";
export const STORE_KEYCLOAK_DATA = "STORE_KEYCLOAK_DATA";
export const STORE_IS_PROPONENT = "STORE_IS_PROPONENT";

// Modals
export const OPEN_MODAL = "OPEN_MODAL";
export const CLOSE_MODAL = "CLOSE_MODAL";
export const CHANGE_MODAL_TITLE = "CHANGE_MODAL_TITLE";

// Variances
export const STORE_VARIANCES = "STORE_VARIANCES";
export const STORE_VARIANCE = "STORE_VARIANCE";
export const STORE_COMPLIANCE_CODES = "STORE_COMPLIANCE_CODES";
export const STORE_VARIANCE_STATUS_OPTIONS = "STORE_VARIANCE_STATUS_OPTIONS";
export const STORE_VARIANCE_DOCUMENT_CATEGORY_OPTIONS = "STORE_VARIANCE_DOCUMENT_CATEGORY_OPTIONS";

// Reports
export const STORE_MINE_REPORTS = "STORE_MINE_REPORTS";
