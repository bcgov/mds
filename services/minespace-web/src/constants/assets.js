// all static files
export { default as BC_GOV } from "../assets/images/gov_bc_logo.png";
export { default as RED_CLOCK } from "../assets/images/late-clock-red.svg";
export { default as WARNING } from "../assets/images/warning.svg";
export { default as MENU } from "../assets/images/Menu.svg";
export { default as SINGLE_DOCUMENT } from "../assets/images/single-document.svg";
export { default as DOCUMENTS } from "../assets/images/documents.svg";
export { default as EDIT_PENCIL } from "../assets/images/edit-pencil.svg";

export {
  default as USER_GUIDE,
} from "../assets/downloads/User Guide - Register of Tailings Storage Facilities and Dams.pdf";
export {
  default as TAILINGS_DOCUMENT,
} from "../assets/downloads/Register of Tailings Storage Facilities and Dams.xlsm";
