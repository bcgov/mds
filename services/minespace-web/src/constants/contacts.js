class Contact {
  constructor(title, name, phone, email) {
    this.title = title;
    this.name = name;
    this.phone = phone;
    this.email = email;
  }
}

class RegionalMineRegionalContacts {
  constructor(safety, permitting, director, office) {
    this.safety = safety;
    this.permitting = permitting;
    this.director = director;
    this.office = office;
  }
}

class MajorMineRegionalContacts {
  constructor(health) {
    this.health = health;
  }
}

const TITLE_SAFETY = "Senior Health, Safety and Environment Inspector";
const TITLE_PERMITTING = "Senior Permitting Inspector";
const TITLE_DIRECTOR = "Regional Director";
const TITLE_OFFICE = "Regional Office";

const REG_NE = new RegionalMineRegionalContacts(
  new Contact(TITLE_SAFETY, "Brian Oke", "(250) 565-4387", "Brian.Oke@gov.bc.ca"),
  new Contact(TITLE_PERMITTING, "Marnie Fraser", "(250) 565-4206", "Marnie.Fraser@gov.bc.ca"),
  new Contact(TITLE_DIRECTOR, "Victor Koyanagi", "(250) 565-4323", "Victor.Koyanagi@gov.bc.ca"),
  new Contact(TITLE_OFFICE, null, "(250) 565-4240", "MMD-PrinceGeorge@gov.bc.ca")
);

const REG_NW = new RegionalMineRegionalContacts(
  new Contact(TITLE_SAFETY, "Doug Flynn", "(250) 847-7386", "Doug.Flynn@gov.bc.ca"),
  new Contact(TITLE_PERMITTING, "Andrea Ross", "(250) 847-7768", "Andrea.Ross@gov.bc.ca"),
  new Contact(TITLE_DIRECTOR, "Howard Davies", "(250) 847-7653", "Howard.Davies@gov.bc.ca"),
  new Contact(TITLE_OFFICE, null, "(250) 847-7383", "MMD-Smithers@gov.bc.ca")
);

const REG_SC = new RegionalMineRegionalContacts(
  new Contact(TITLE_SAFETY, "Chris LeClair", "(250) 371-3714", "Chris.LeClair@gov.bc.ca"),
  new Contact(TITLE_PERMITTING, null, "(250) 371-3912", "MMD-Kamloops@gov.bc.ca"),
  new Contact(TITLE_DIRECTOR, "Rick Adams", "(250) 828-4583", "Rick.Adams@gov.bc.ca"),
  new Contact(TITLE_OFFICE, null, "(250) 371-3912", "MMD-Kamloops@gov.bc.ca")
);

const REG_SE = new RegionalMineRegionalContacts(
  new Contact(TITLE_SAFETY, "Michael Daigle", "(250) 417-6141", "Michael.Daigle@gov.bc.ca"),
  new Contact(TITLE_PERMITTING, "Glen Hendrickson", "(250) 417-6033", "Glen.Hendrickson@gov.bc.ca"),
  new Contact(TITLE_DIRECTOR, "Kathie Wagar", "(250) 417-6011", "Kathie.Wagar@gov.bc.ca"),
  new Contact(TITLE_OFFICE, null, "(250) 417-6134", "MMD-Cranbrook@gov.bc.ca")
);

const REG_SW = new RegionalMineRegionalContacts(
  new Contact(TITLE_SAFETY, "Jim Dunkley", "(778) 698-7294", "Jim.Dunkley@gov.bc.ca"),
  new Contact(TITLE_PERMITTING, "Don Harrison", "(778) 698-7014", "Donald.Harrison@gov.bc.ca"),
  new Contact(TITLE_DIRECTOR, "Matthew McLean", "(778) 698-9411", "Matthew.MacLean@gov.bc.ca"),
  new Contact(TITLE_OFFICE, null, "(778) 698-3649", "SouthwestMinesDivision@gov.bc.caâ€‹")
);

export const REGIONAL_MINE_REGIONAL_CONTACTS = {
  NE: REG_NE,
  NW: REG_NW,
  SC: REG_SC,
  SE: REG_SE,
  SW: REG_SW,
};

const TITLE_HEALTH = "Health and Safety Inspector";

const MAJ_NE = new MajorMineRegionalContacts(
  new Contact(TITLE_HEALTH, "Kristopher Bailey", "(250) 565-4271", "Kristopher.Bailey@gov.bc.ca")
);

const MAJ_NW = new MajorMineRegionalContacts(
  new Contact(TITLE_HEALTH, "Megan Frederick", "(250) 847-7214", "Megan.Frederick@gov.bc.ca")
);

const MAJ_SC = new MajorMineRegionalContacts(
  new Contact(TITLE_HEALTH, "Blythe Golobic", "(250) 371-3915", "Blythe.Golobic@gov.bc.ca")
);

const MAJ_SE = new MajorMineRegionalContacts(
  new Contact(TITLE_HEALTH, "Alan Day", "(250) 417-6013", "Alan.Day@gov.bc.ca")
);

const MAJ_SW = new MajorMineRegionalContacts(
  new Contact(TITLE_HEALTH, "Blythe Golobic", "(250) 371-3915", "Blythe.Golobic@gov.bc.ca")
);

export const MAJOR_MINE_REGIONAL_CONTACTS = {
  NE: MAJ_NE,
  NW: MAJ_NW,
  SC: MAJ_SC,
  SE: MAJ_SE,
  SW: MAJ_SW,
};

export const MM_OFFICE = new Contact("Major Mines Office", null, null, "PermRecl@gov.bc.ca");

export const CHIEF_INSPECTOR = new Contact(
  "Chief Inspector of Mines",
  "Hermanus Henning",
  "(778) 974-5980",
  "hermanus.henning@gov.bc.ca"
);

export const EXEC_LEAD_AUTH = new Contact(
  "Executive Lead (Authorizations)",
  "George Warnock",
  "(250) 649-4339",
  "brittanytownsend@gov.bc.ca"
);
