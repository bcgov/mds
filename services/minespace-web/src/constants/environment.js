// environment config variables for test/dev/prod
export const DEFAULT_ENVIRONMENT = {
  apiUrl: "http://localhost:5000",
  docManUrl: "http://localhost:5001",
  keycloak_resource: "mines-application-local",
  keycloak_clientId: "minespace-local",
  keycloak_idpHint: "local",
  keycloak_url: "https://sso-test.pathfinder.gov.bc.ca/auth",
  siteminder_url: "https://logontest.gov.bc.ca",
};

const WINDOW_LOCATION = `${window.location.origin}${process.env.BASE_PATH}`;
export const BCEID_LOGIN_REDIRECT_URI = `${WINDOW_LOCATION}/return-page?type=login`;
export const KEYCLOAK_LOGOUT_REDIRECT_URI = `${WINDOW_LOCATION}/return-page?type=logout`;
export const SITEMINDER_LOGOUT_REDIRECT_URI = `${WINDOW_LOCATION}/return-page?type=smlogout&retnow=1`;
