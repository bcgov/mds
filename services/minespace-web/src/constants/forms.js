// Redux Forms
export const ADD_REPORT = "ADD_REPORT";
export const EDIT_REPORT = "EDIT_REPORT";
export const ADD_VARIANCE = "ADD_VARIANCE";
export const EDIT_VARIANCE = "EDIT_VARIANCE";
