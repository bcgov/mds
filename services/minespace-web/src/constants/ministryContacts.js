// Ministry Contacts
export const MINISTRY_CONTACTS = {
  REGIONAL_HEALTH_AND_SAFETY: {
    NAME: "John Doe",
    PHONE: "(250) 479-2546",
    EMAIL: "john.doe@gov.bc.ca",
  },
  REGIONAL_MINE_GENERAL: {
    EMAIL: "inforegionalmines@gov.bc.ca",
  },
  MAJOR_MINE_GENERAL: {
    EMAIL: "infomajormines@gov.bc.ca",
  },
};

export default MINISTRY_CONTACTS;
