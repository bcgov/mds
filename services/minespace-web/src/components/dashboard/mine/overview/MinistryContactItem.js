import React from "react";
import PropTypes from "prop-types";
import { Typography } from "antd";
import * as Strings from "@/constants/strings";

const { Paragraph, Text } = Typography;

const propTypes = {
  contact: PropTypes.objectOf(PropTypes.any).isRequired,
};

export const MinistryContactItem = (props) => (
  <Paragraph className="ministry-contact-item">
    <Text strong className="ministry-contact-title">
      {props.contact.title || Strings.UNKNOWN}
      <br />
    </Text>
    {props.contact.name && (
      <Text>
        {props.contact.name}
        <br />
      </Text>
    )}
    {props.contact.phone && (
      <Text>
        {props.contact.phone}
        <br />
      </Text>
    )}
    {props.contact.email && (
      <Text>
        <a href={`mailto:${props.contact.email}`}>{props.contact.email}</a>
        <br />
      </Text>
    )}
  </Paragraph>
);

MinistryContactItem.propTypes = propTypes;

export default MinistryContactItem;
