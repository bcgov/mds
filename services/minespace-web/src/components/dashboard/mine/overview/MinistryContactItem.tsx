import React from "react";
import { connect } from "react-redux";
import { Typography } from "antd";
import { getEMLIContactTypesHash } from "@mds/common/redux/selectors/staticContentSelectors";
import * as Strings from "@/constants/strings";

interface MinistryContactItemProps {
  contact: {
    emli_contact_type_code: string;
    first_name: string;
    last_name: string;
    phone_number: string;
    email: string;
  };
  EMLIContactTypesHash: {
    [key: string]: string;
  };
}

export const MinistryContactItem: React.FC<MinistryContactItemProps> = (
  props: MinistryContactItemProps
) => (
  <Typography.Paragraph className="ministry-contact-item">
    <Typography.Text strong className="ministry-contact-title">
      {props.EMLIContactTypesHash[props.contact.emli_contact_type_code] || Strings.UNKNOWN}
      <br />
    </Typography.Text>
    {props.contact.first_name && props.contact.last_name && (
      <Typography.Text>
        {props.contact.first_name} {props.contact.last_name}
        <br />
      </Typography.Text>
    )}
    {props.contact.phone_number && (
      <Typography.Text>
        {props.contact.phone_number}
        <br />
      </Typography.Text>
    )}
    {props.contact.email && (
      <Typography.Text>
        <a href={`mailto:${props.contact.email}`}>{props.contact.email}</a>
        <br />
      </Typography.Text>
    )}
  </Typography.Paragraph>
);

const mapStateToProps = (state) => ({
  EMLIContactTypesHash: getEMLIContactTypesHash(state),
});

export default connect(mapStateToProps)(MinistryContactItem);
