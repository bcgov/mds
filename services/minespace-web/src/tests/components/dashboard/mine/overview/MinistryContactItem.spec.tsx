import React from "react";
import { render } from "@testing-library/react";
import MinistryContactItem from "@/components/dashboard/mine/overview/MinistryContactItem";
import { BULK_STATIC_CONTENT_RESPONSE } from "@mds/common/tests/mocks/dataMocks";
import { ReduxWrapper } from "@/tests/utils/ReduxWrapper";
import { STATIC_CONTENT } from "@mds/common/constants/reducerTypes";

const initialState = {
  [STATIC_CONTENT]: BULK_STATIC_CONTENT_RESPONSE,
};

const setupProps = () => ({
  contact: {
    emli_contact_type_code: "SPI",
    first_name: "John",
    last_name: "Doe",
    phone_number: "123-456-7890",
    email: "johndoe@example.com",
  },
});

describe("MinistryContactItem", () => {
  it("renders properly", () => {
    const wrapper = render(
      <ReduxWrapper initialState={initialState}>
        <MinistryContactItem {...setupProps()} />
      </ReduxWrapper>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
