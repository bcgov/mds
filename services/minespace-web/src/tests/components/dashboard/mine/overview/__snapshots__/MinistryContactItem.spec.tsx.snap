// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinistryContactItem renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-typography ministry-contact-item"
      >
        <span
          class="ant-typography ministry-contact-title"
        >
          <strong>
            Senior Permitting Inspector
            <br />
          </strong>
        </span>
        <span
          class="ant-typography"
        >
          John
           
          Doe
          <br />
        </span>
        <span
          class="ant-typography"
        >
          123-456-7890
          <br />
        </span>
        <span
          class="ant-typography"
        >
          <a
            href="mailto:johndoe@example.com"
          >
            johndoe@example.com
          </a>
          <br />
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-typography ministry-contact-item"
    >
      <span
        class="ant-typography ministry-contact-title"
      >
        <strong>
          Senior Permitting Inspector
          <br />
        </strong>
      </span>
      <span
        class="ant-typography"
      >
        John
         
        Doe
        <br />
      </span>
      <span
        class="ant-typography"
      >
        123-456-7890
        <br />
      </span>
      <span
        class="ant-typography"
      >
        <a
          href="mailto:johndoe@example.com"
        >
          johndoe@example.com
        </a>
        <br />
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
